src/simulation/Benchmark.cpp

```diff
@@ -0,0 +1,402 @@
+    : mInitialize(false)
+    , mPopulate(false)
+    , mTxRate(0)
+    , mAccounts(0)
+    , mNetworkID(networkID)
+{
+}
+
+Benchmark::BenchmarkBuilder&
+Benchmark::BenchmarkBuilder::setNumberOfInitialAccounts(uint32_t accounts)
+{
+    mAccounts = accounts;
+    return *this;
+}
+
+Benchmark::BenchmarkBuilder&
+Benchmark::BenchmarkBuilder::setTxRate(uint32_t txRate)
+{
+    mTxRate = txRate;
+    return *this;
+}
+
+Benchmark::BenchmarkBuilder&
+Benchmark::BenchmarkBuilder::initializeBenchmark()
+{
+    mInitialize = true;
+    return *this;
+}
+
+Benchmark::BenchmarkBuilder&
+Benchmark::BenchmarkBuilder::populateBenchmarkData()
+{
+    mPopulate = true;
+    return *this;
+}
+
+std::unique_ptr<Benchmark>
+Benchmark::BenchmarkBuilder::createBenchmark(Application& app) const
+{
+    auto sampler = make_unique<ShuffleLoadGenerator>(mNetworkID);
+    if (mPopulate)
+    {
+        prepareBenchmark(app, *sampler);
+    }
+    if (mInitialize)
+    {
+        sampler->initialize(app, mAccounts);
+    }
+    setMaxTxSize(app.getLedgerManager(), MAXIMAL_NUMBER_OF_TXS_PER_LEDGER);
+    app.getHerder().triggerNextLedger(app.getLedgerManager().getLedgerNum());
+    app.getHistoryManager().queueCurrentHistory();
+
+    class BenchmarkExt : public Benchmark
+    {
+      public:
+        BenchmarkExt(medida::MetricsRegistry& registry, uint32_t txRate,
+                     std::unique_ptr<TxSampler> sampler)
+            : Benchmark(registry, txRate, std::move(sampler))
+        {
+        }
+    };
+    return make_unique<BenchmarkExt>(
+        app.getMetrics(), mTxRate,
+        std::unique_ptr<TxSampler>(sampler.release()));
+}
+
+void
+Benchmark::BenchmarkBuilder::prepareBenchmark(
+    Application& app, ShuffleLoadGenerator& sampler) const
+{
+    populateAccounts(app, mAccounts, sampler);
+}
+
+void
+Benchmark::BenchmarkBuilder::populateAccounts(
+    Application& app, size_t size, ShuffleLoadGenerator& sampler) const
+{
+    for (size_t accountsLeft = size, batchSize = size; accountsLeft > 0;
+         accountsLeft -= batchSize)
+    {
+        batchSize = std::min(accountsLeft, MAXIMAL_NUMBER_OF_TXS_PER_LEDGER);
+        auto newAccounts = sampler.createAccounts(batchSize);
+        createAccountsDirectly(app, newAccounts);
+    }
+}
+
+void
+Benchmark::BenchmarkBuilder::setMaxTxSize(LedgerManager& ledger,
+                                          uint32_t maxTxSetSize) const
+{
+    StellarValue sv(ledger.getLastClosedLedgerHeader().hash,
+                    ledger.getLedgerNum(), emptyUpgradeSteps, 0);
+    {
+        LedgerUpgrade up(LEDGER_UPGRADE_MAX_TX_SET_SIZE);
+        up.newMaxTxSetSize() = maxTxSetSize;
+        Value v(xdr::xdr_to_opaque(up));
+        sv.upgrades.emplace_back(v.begin(), v.end());
+    }
+    LedgerCloseData ledgerData = createData(ledger, sv);
+    ledger.closeLedger(ledgerData);
+}
+
+LedgerCloseData
+Benchmark::BenchmarkBuilder::createData(LedgerManager& ledger,
```

Comment:
This does not belong here. Check if there is similar method in one of tests namespaces (like testutil), if not, create a free function there.

There is no dependency on benchmark here.